internal/expr/complete.go

```diff
@@ -0,0 +1,49 @@
+package expr
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type CompletedExpr struct {
+	ParsedExpr  *ParsedExpr
+	InputValues []any
+}
+
+type typeNameToValue = map[string]any
+
+// Complete extracts query arguments specified in inputParts from structs.
+func (pe *PreparedExpr) Complete(args ...any) (ce *CompletedExpr, err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("cannot complete expression: %s", err)
+		}
+	}()
+
+	var tv = make(typeNameToValue)
+	for _, arg := range args {
+		if arg == (any)(nil) {
+			return nil, fmt.Errorf("cannot reflect nil value")
+		}
+		tv[reflect.TypeOf(arg).Name()] = arg
+	}
+
+	// Query parameteres.
+	qargs := []any{}
+
+	for _, p := range pe.ParsedExpr.queryParts {
+		if p, ok := p.(*inputPart); ok {
+			v, ok := tv[p.source.prefix]
```

Comment:
I realise this probably isn't something to solve here, but somehow the `prefix` field storing the name of a type feels odd. Perhaps calling this `typeName` would be more natural?

Reply:
Yea, I'm not happy with the names of these either. They were decided on at the start after a lot of back and forth though, so I think we have to wait until that's a distant memory before changing them again.